### ISSUE TITLE
Ducument how best to put new pages behind switches

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#5868]
Ref: https://github.com/mozilla/bedrock/issues/5747#issuecomment-402277888

Occasionally we need to enable new pages in production that are time sensitive, such as the [annual report](https://www.mozilla.org/en-US/foundation/annualreport/2016/) or [transparency report](https://www.mozilla.org/en-US/about/policy/transparency/jul-dec-2017/). In the past we have done this by putting a switch [around a URL](https://github.com/mozilla/bedrock/pull/5848), but now that switches are in the local database this creates some new issues.

My understanding is as follows:

- We're fine to use switches in templates and views as these can read from the local DB.
- We're not OK to use switches in code that requires a server restart, such as `urls.py`.

If the above is correct, it would be good to have the following:

- Some guidelines on where it is safe to use a switch and where it is not.
- A suggested method by wich we can safely put new pages behind a switch.

**Comment by pmac** [comment]
And to be clear, the old way of setting a switch via the [config files that require a deployment](https://github.com/mozmeao/www-config/blob/master/configs/bedrock-prod.env) is still an option. So if you need to use a switch in this way you definitely can, it's just that it needs to go in a different spot. It may be that we just need separate facilities for these things. The other option is to declare the db-based switches a failed experiment and remove that capability. It does introduce some edge cases that are clearly difficult to spot.

**Comment by jgmize** [comment]
I think it would make sense to do a fast global deployment and eliminate the use of switches entirely for this scenario.

**Comment by alexgibson** [comment]
> I think it would make sense to do a fast global deployment and eliminate the use of switches entirely for this scenario.

While we don’t need to do timed releases regularly, i’m not sure this will really be adequate enough if something needs to happen for a specific time on a given day.

**Comment by pmac** [comment]
Deployments can be blocked by a large number of things still, and these changes can be needed when SREs are sleeping. This scenario is what switches are good at. I think the answer is probably just to go ahead and add the new page and URL and use the switch on the page to change the links. Would that be acceptable to the team @alexgibson? The page would technically be "live" at deployment, but wouldn't be discoverable until the switch was flipped.

**Comment by alexgibson** [comment]
> I think the answer is probably just to go ahead and add the new page and URL and use the switch on the page to change the links.

I think this would be fine for pages that don't contain sensitive information, but for pages used by the press who actively look for URLs (such as annual report), I'm not sure if hoping no-one discovers the URL would be enough.

It's probably also worth keeping in mind that these requests only happen a couple of times per year. It's not something worth spending a whole lot of engineering time on, but having some recommended way to facilitate it (even if a bit clumbersome) would be good still.

**Comment by alexgibson** [comment]
Thanks, @pmac 👍

**Comment by jgmize** [comment]
@pmac pointed out another issue with putting pages behind SQLite “DB” based switches in #websites on the Mozilla slack earlier today:

> here’s my guess: when a new instance of the server starts up it has a copy of the database from when it was built, and since we haven’t deployed in a while that database does not have that switch flipped

New containers are launched all the time in response to autoscaling events as load varies, so a deployment is needed to ensure that switches stay flipped.
